### PR TITLE
Separate middle name and suffix in the sign up flow (MNTOR-3521)

### PR DESCRIPTION
**Pull request by flozia**
# References:

Jira: [MNTOR-3521](https://mozilla-hub.atlassian.net/browse/MNTOR-3521)

# Description

Add the ability to run experiments on the optional “middle name” and “suffix” fields in different combinations.

# Screenshot 

| Middle name only | Suffix only | Middle name and suffix |
| --- | --- | --- |
| image | image | image |

# Checklist (Definition of Done)

- [x] Commits in this PR are minimal and [have descriptive commit messages](https://chris.beams.io/posts/git-commit/).
- [x] Product Owner accepted the User Story (demo of functionality completed) or waived the privilege.
- [x] All acceptance criteria are met.

[MNTOR-3521]: https://mozilla-hub.atlassian.net/browse/MNTOR-3521?atlOrigin=eyJpIjoiNWRkNTljNzYxNjVmNDY3MDlhMDU5Y2ZhYzA5YTRkZjUiLCJwIjoiZ2l0aHViLWNvbS1KU1cifQ

### CODE REVIEW

File: config/nimbus.yaml
Comment (Vinnl) [inline config/nimbus.yaml:1]: I'm assuming these can be updated and that that will be picked up correctly by Nimbus, @rhelmer?

File: src/app/(proper_react)/(redesign)/(authenticated)/user/welcome/EnterInfo.module.scss [hunk 1]
```diff
@@ -84,8 +84,9 @@
   padding: $spacing-lg 0;
 
   @media screen and (min-width: $screen-lg) {
-    // The child items in the grid are input fields. Items marked as “optional”
-    // are only shown if the experiment `welcome-scan-optional-info` is enabled.
+    // The child items in the grid are input fields. If items marked as
```
Comment (Vinnl) [inline src/app/(proper_react)/(redesign)/(authenticated)/user/welcome/EnterInfo.module.scss:87]: Supernit:

```suggestion
    // The child items in the grid are input fields. Whether items marked as
```

File: src/app/(proper_react)/(redesign)/(authenticated)/user/welcome/EnterInfo.tsx [hunk 2]
```diff
@@ -193,7 +194,19 @@ export const EnterInfo = ({
...
+      experimentData["welcome-scan-optional-info"];
+    const showOptionalInfo =
+      optionalInfoExperimentData.enabled &&
+      ((userDetail.key === "middle_name" &&
+        (optionalInfoExperimentData.variant === "middleName" ||
+          optionalInfoExperimentData.variant === "suffixAndMiddleName")) ||
+        (userDetail.key === "name_suffix" &&
+          (optionalInfoExperimentData.variant === "suffix" ||
+            optionalInfoExperimentData.variant === "suffixAndMiddleName")));
+    return userDetail.isRequired || showOptionalInfo;
```
Comment (Vinnl) [inline src/app/(proper_react)/(redesign)/(authenticated)/user/welcome/EnterInfo.tsx:208]: Optional suggestion, to reduce the large number of nested conjunctions/disjunctions - but not actually sure if this is clearer:

```suggestion
    if (userDetail.isRequired) {
      return true;
    }

    const optionalInfoExperimentData =
      experimentData["welcome-scan-optional-info"];
    if(optionalInfoExperimentData.enabled) {
      if (userDetail.key === "middle_name") {
        return  (optionalInfoExperimentData.variant === "middleName" ||
          optionalInfoExperimentData.variant === "suffixAndMiddleName");
      }
      if (userDetail.key === "name_suffix") {
        return  (optionalInfoExperimentData.variant === "suffix" ||
          optionalInfoExperimentData.variant === "suffixAndMiddleName");
      }
    }
    return false;
```

**Review by Vinnl — APPROVED** [review]
Code looks good!

File: src/app/(proper_react)/(redesign)/(authenticated)/user/welcome/EnterInfo.tsx (hunk 2 above)
Comment (flozia) [reply src/app/(proper_react)/(redesign)/(authenticated)/user/welcome/EnterInfo.tsx]: I like the suggestion — it is easier to dive in for someone reading it and the early return makes sense as well. Updated in [64d7693](https://github.com/mozilla/blurts-server/pull/5035/commits/64d769351089c3178ba14523359d35856fc4a2e2).

**Pull request merged**